internal/notifier/azure_eventhub.go

```diff
@@ -116,3 +132,42 @@ func newSASHub(address string) (*eventhub.Hub, error) {
 
 	return hub, nil
 }
+
+// newManagedIdentityToken is used to attempt credential-free authentication.
+func newManagedIdentityToken(ctx context.Context, proxy, serviceAccountName, providerName, providerNamespace string, tokenClient client.Client, tokenCache *pkgcache.TokenCache) (string, error) {
+	var opts []pkgauth.Option
+
+	if proxy != "" {
+		proxyURL, err := url.Parse(proxy)
+		if err != nil {
+			return "", fmt.Errorf("error parsing proxy URL : %w", err)
+		}
+		opts = append(opts, pkgauth.WithProxyURL(*proxyURL))
+	}
+
+	if serviceAccountName != "" {
+		opts = append(opts, pkgauth.WithServiceAccount(types.NamespacedName{
+			Name:      serviceAccountName,
+			Namespace: providerNamespace,
+		}, tokenClient))
+	}
+
+	if tokenCache != nil {
+		involvedObject := cache.InvolvedObject{
+			Kind:      v1beta3.ProviderKind,
+			Name:      providerName,
+			Namespace: providerNamespace,
+		}
+		opts = append(opts, pkgauth.WithCache(*tokenCache, involvedObject))
+	}
+
+	tokenCredential := azure.NewTokenCredential(ctx, opts...)
+	accessToken, err := tokenCredential.GetToken(ctx, policy.TokenRequestOptions{
+		Scopes: []string{azure.ScopeEventHubs},
+	})
+	if err != nil {
+		return "", fmt.Errorf("failed to get token for azure event hub scope %v", err)
+	}
+
+	return accessToken.Token, nil
```

Comment:
Since here you need the token right away (as opposed to passing an `azcore.TokenCredential` to an Azure library downstream) you don't need to use `azure.NewTokenCredential()`, you can use `pkgauth.GetToken()` directly.

Also, let's never pass scopes through `policy.TokenRequestOptions`, we have `pkgauth.WithScopes()` for that. We only accept scopes through `policy.TokenRequestOptions` in `azure.TokenCredential.GetToken()` because Azure downstream libraries receiving an `azcore.TokenCredential` need it.

So this call should look like this:

```suggestion
	token, err := pkgauth.GetToken(ctx, azure.Provider{}, opts...)
	if err != nil {
		return "", fmt.Errorf("failed to get token for azure event hub: %w", err)
	}

	return token.(*azure.Token).AccessToken.Token, nil
```